cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowContext.java

```diff
@@ -27,24 +27,33 @@
 
   WorkflowSpecification getWorkflowSpecification();
 
+  /**
+   * @throws UnsupportedOperationException if it is called from {@link Predicate}
+   */
   WorkflowActionSpecification getSpecification();
 
   long getLogicalStartTime();
 
   /**
-   * Returns a {@link Callable} that launches the associated program
-   * with the specified name when the {@link Callable#call() call} method is invoked. When the program completes,
-   * the {@link Callable} returns the {@link RuntimeContext} of the program or {@code null} if
+   * Returns a {@link Callable} that launches the associated program with the specified name when
+   * the {@link Callable#call() call} method is invoked. When the program completes,
+   * the {@link Callable} returns the {@link WorkflowToken} for the program run or {@code null} if
    * no such context exists.
    * <p/>
    * <p> An Exception is thrown from the {@link Callable#call()} method if the program fails </p>
    *
    * @throws IllegalArgumentException if no program with the specified name is defined in the workflow
+   * @throws UnsupportedOperationException if it is called from {@link Predicate}
    */
-  Callable<RuntimeContext> getProgramRunner(String name);
+  Callable<WorkflowToken> getProgramRunner(String name);
```

Comment:
I don't understand. Before, the program runner had access to the runtime arguments. Now it does not? How will runtime arguments be passed to programs then?